src/Datasource/EntityTrait.php

```diff
@@ -891,6 +891,31 @@ public function isNew($new = null)
         return $this->_new = $new;
     }
 
+    /**
+     * Returns whether this entity has errors.
+     *
+     * @param bool $includeHasErrorsFromNestedEntities
+     * @return bool
+     */
+    public function hasErrors($includeHasErrorsFromNestedEntities = true)
+    {
+        if (empty($this->_errors) === false) {
```

Comment:
This can be written as `if (!empty($this->_errors)) {`. Similarly all other `if` statements where you compare a return value or argument which is already boolean should be updated.